### PR TITLE
Cleanup DLP Risk Analysis snippets

### PR DESCRIPTION
**Pull request by shubha-rajan**
Working towards #1479
This ended up being a larger PR than expected. I can split it up if needed.

### CODE REVIEW

**Comment by lesv** [comment]
Please fix compile problems

**Comment by shubha-rajan** [comment]
> Please fix compile problems
build failures are a result of broken tests that were removed in $2062 and #2065. Once those are merged in, everything should pass.

File: dlp/src/main/java/dlp/snippets/RiskAnalysisKMap.java [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+            DlpJob dlpJob = dlpServiceClient.createDlpJob(createDlpJobRequest);
+
+            // Set up a Pub/Sub subscriber to listen on the job completion status
+            final SettableApiFuture<Boolean> done = SettableApiFuture.create();
+
+            ProjectSubscriptionName subscriptionName =
+                    ProjectSubscriptionName.of(projectId, subscriptionId);
+
+            MessageReceiver handleMessage =
+                    (PubsubMessage pubsubMessage, AckReplyConsumer ackReplyConsumer) -> {
```
Comment (kurtisvg) [inline dlp/src/main/java/dlp/snippets/RiskAnalysisKMap.java]: Long lamba again

File: dlp/src/main/java/dlp/snippets/RiskAnalysisKMap.java [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+                            GetDlpJobRequest.newBuilder()
+                                    .setName(dlpJob.getName())
+                                    .build());
+            System.out.println("Job status: " + completedJob.getState());
+
+            // Get the result and parse through and process the information
+            KMapEstimationResult kmapResult  = completedJob.getRiskDetails().getKMapEstimationResult();
+
+            for (KMapEstimationHistogramBucket result : kmapResult.getKMapEstimationHistogramList()) {
+
```
Comment (kurtisvg) [inline dlp/src/main/java/dlp/snippets/RiskAnalysisKMap.java]: nit: no newline

File: dlp/src/main/java/dlp/snippets/RiskAnalysisKMap.java [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+                        "\tAnonymity range: [%d, %d]\n", result.getMinAnonymity(), result.getMaxAnonymity());
+                System.out.printf("\tSize: %d\n", result.getBucketSize());
+
+                for (KMapEstimationQuasiIdValues valueBucket : result.getBucketValuesList()) {
+                    String quasiIdValues =
+                            valueBucket
+                                    .getQuasiIdsValuesList()
+                                    .stream()
+                                    .map(
+                                            v -> {
```
Comment (kurtisvg) [inline dlp/src/main/java/dlp/snippets/RiskAnalysisKMap.java]: A comment on this function would be helpful.

nit: Does there have to be a new line before `v -> {` (might be a java format thing, just looks weird to me)?

File: dlp/src/main/java/dlp/snippets/RiskAnalysisLDiversity.java [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+
+    }
+    public static void calculateLDiversity(
+            String projectId,
+            String datasetId,
+            String tableId,
+            String topicId,
+            String subscriptionId,
+            String sensitiveAttribute,
+            List<String> quasiIds) throws Exception {
```
Comment (kurtisvg) [inline dlp/src/main/java/dlp/snippets/RiskAnalysisLDiversity.java]: sensitiveAttribute, quasiIds inlined

File: dlp/src/main/java/dlp/snippets/RiskAnalysisLDiversity.java [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+            DlpJob dlpJob = dlpServiceClient.createDlpJob(createDlpJobRequest);
+
+            // Set up a Pub/Sub subscriber to listen on the job completion status
+            final SettableApiFuture<Boolean> done = SettableApiFuture.create();
+
+            ProjectSubscriptionName subscriptionName =
+                    ProjectSubscriptionName.of(projectId, subscriptionId);
+
+            MessageReceiver handleMessage =
+                    (PubsubMessage pubsubMessage, AckReplyConsumer ackReplyConsumer) -> {
```
Comment (kurtisvg) [inline dlp/src/main/java/dlp/snippets/RiskAnalysisLDiversity.java]: Long lambda

File: dlp/src/main/java/dlp/snippets/RiskAnalysisCategoricalStats.java [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+            DlpJob dlpJob = dlpServiceClient.createDlpJob(createDlpJobRequest);
+
+            // Set up a Pub/Sub subscriber to listen on the job completion status
+            final SettableApiFuture<Boolean> done = SettableApiFuture.create();
+
+            ProjectSubscriptionName subscriptionName =
+                    ProjectSubscriptionName.of(projectId, subscriptionId);
+
+            MessageReceiver handleMessage =
+                    (PubsubMessage pubsubMessage, AckReplyConsumer ackReplyConsumer) -> {
```
[the remark this answers is not in the document]
Comment (shubha-rajan) [reply dlp/src/main/java/dlp/snippets/RiskAnalysisCategoricalStats.java]: I was unsure of how to do this when the lambda used variables outside of the scope of the function, like `dlpJob` and `done`. `handleMessage` needs to be callable so I need to return a function. I'm not sure if there's a way to build functions using closures in Java? I ended up nesting the lambda within another function, and passing the `DlpJob` and `Future` to that function in in c634313, but that seems more complicated and less readable to me? Thoughts?

File: dlp/src/main/java/dlp/snippets/RiskAnalysisCategoricalStats.java [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+                done.get(1, TimeUnit.MINUTES);
+                Thread.sleep(500); // Wait for the job to become available
+            } catch (TimeoutException e) {
+                System.out.println("Unable to verify job completion.");
+            }
+
+            // Retrieve completed job status
+            DlpJob completedJob =
+                    dlpServiceClient.getDlpJob(
+                            GetDlpJobRequest.newBuilder()
```
[the remark this answers is not in the document]
Comment (shubha-rajan) [reply dlp/src/main/java/dlp/snippets/RiskAnalysisCategoricalStats.java]: resolved in 8546690

File: dlp/src/main/java/dlp/snippets/RiskAnalysisCategoricalStats.java [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+                                    .setName(dlpJob.getName())
+                                    .build());
+            System.out.println("Job status: " + completedJob.getState());
+
+            // Get the result and parse through and process the information
+            CategoricalStatsResult result =
+                    completedJob.getRiskDetails().getCategoricalStatsResult();
+
```
[the remark this answers is not in the document]
Comment (shubha-rajan) [reply dlp/src/main/java/dlp/snippets/RiskAnalysisCategoricalStats.java]: Resolved

File: dlp/src/main/java/dlp/snippets/RiskAnalysisKAnonymity.java [hunk 9]
```diff
@@ -0,0 +1,183 @@
...
+class RiskAnalysisKAnonymity {
+
+    public static void calculateKAnonymity() throws Exception {
+        // TODO(developer): Replace these variables before running the sample.
+        String projectId = "your-project-id";
+        String datasetId = "your-bigquery-dataset-id";
+        String tableId = "your-bigquery-table-id";
+        String topicId = "pub-sub-topic";
+        String subscriptionId = "pub-sub-subscription";
+        List<String> quasiIdColumns = Arrays.asList("name", "age", "zip_code", "...");
```
[the remark this answers is not in the document]
Comment (shubha-rajan) [reply dlp/src/main/java/dlp/snippets/RiskAnalysisKAnonymity.java]: Resolved in 8546690

File: dlp/src/main/java/dlp/snippets/RiskAnalysisKAnonymity.java [hunk 10]
```diff
@@ -0,0 +1,183 @@
...
+                            GetDlpJobRequest.newBuilder()
+                                    .setName(dlpJob.getName())
+                                    .build());
+            System.out.println("Job status: " + completedJob.getState());
+
+
+            KAnonymityResult kanonymityResult =
+                    completedJob.getRiskDetails().getKAnonymityResult();
+
+            // Get the result and parse through and process the information
```
[the remark this answers is not in the document]
Comment (shubha-rajan) [reply dlp/src/main/java/dlp/snippets/RiskAnalysisKAnonymity.java]: resolved

File: dlp/src/main/java/dlp/snippets/RiskAnalysisKMap.java [hunk 11]
```diff
@@ -0,0 +1,223 @@
...
+            String subscriptionId,
+            List<String> quasiIds,
+            List<String> infoTypeNames) throws Exception {
+
+        // Initialize client that will be used to send requests. This client only needs to be created
+        // once, and can be reused for multiple requests. After completing all of your requests, call
+        // the "close" method on the client to safely clean up any remaining background resources.
+        try (DlpServiceClient dlpServiceClient = DlpServiceClient.create()) {
+
+
```
[the remark this answers is not in the document]
Comment (shubha-rajan) [reply dlp/src/main/java/dlp/snippets/RiskAnalysisKMap.java]: all of these were resolved in 82c1f07 after running a code formatter

File: dlp/src/main/java/dlp/snippets/RiskAnalysisLDiversity.java (hunk 4 above)
Comment (shubha-rajan) [reply dlp/src/main/java/dlp/snippets/RiskAnalysisLDiversity.java]: resolved in 82c1f07

File: dlp/src/main/java/dlp/snippets/RiskAnalysisKMap.java (hunk 2 above)
Comment (shubha-rajan) [reply dlp/src/main/java/dlp/snippets/RiskAnalysisKMap.java]: resolved in 92ea643

File: dlp/src/main/java/dlp/snippets/RiskAnalysisLDiversity.java (hunk 5 above)
Comment (shubha-rajan) [reply dlp/src/main/java/dlp/snippets/RiskAnalysisLDiversity.java]: See my previous comment about the lambda. Also, this lambda is also used in the Inspect snippets, so if we change it here, we should probably change it there as well.

File: dlp/src/main/java/dlp/snippets/RiskAnalysisKMap.java (hunk 3 above)
Comment (shubha-rajan) [reply dlp/src/main/java/dlp/snippets/RiskAnalysisKMap.java]: fixed the weird formatting thing, and also rewrote this part to be a bit clearer and more consistent with other samples in e6d4f91. I can also go in and comment on it if that's still necessary.

File: dlp/src/main/java/dlp/snippets/RiskAnalysisKMap.java [hunk 12]
```diff
@@ -0,0 +1,223 @@
...
+class RiskAnalysisKMap {
+
+    public static void calculateKMap() throws Exception {
+        // TODO(developer): Replace these variables before running the sample.
+        String projectId = "your-project-id";
+        String datasetId = "your-bigquery-dataset-id";
+        String tableId = "your-bigquery-table-id";
+        String topicId = "pub-sub-topic";
+        String subscriptionId = "pub-sub-subscription";
+        List<String> quasiIdColumns = Arrays.asList("age", "gender");
```
[the remark this answers is not in the document]
Comment (shubha-rajan) [reply dlp/src/main/java/dlp/snippets/RiskAnalysisKMap.java]: fixed in e6d4f91

File: dlp/src/main/java/dlp/snippets/RiskAnalysisCategoricalStats.java [hunk 13]
```diff
@@ -0,0 +1,169 @@
...
+            for (CategoricalStatsHistogramBucket bucket :
```
[the remark this answers is not in the document]
Comment (shubha-rajan) [reply dlp/src/main/java/dlp/snippets/RiskAnalysisCategoricalStats.java]: fixed in e6d4f91

**Comment by shubha-rajan** [comment]
Presubmit tests are going to fail because of linting errors in other files outside of this PR. Those fixes are handled in #2083, since we probably don't want all of those changes in this PR

File: dlp/src/main/java/dlp/snippets/RiskAnalysisKMap.java [hunk 14]
```diff
@@ -0,0 +1,212 @@
...
+          infoTypeNames.stream()
+              .map(it -> InfoType.newBuilder().setName(it).build())
+              .collect(Collectors.toList());
+
+      if (quasiIds.size() != infoTypes.size()) {
+        throw new IllegalArgumentException("The numbers of quasi-IDs and infoTypes must be equal!");
+      }
+
+      ArrayList<TaggedField> taggedFields =
+          IntStream.range(0, quasiIds.size())
```
[the remark this answers is not in the document]
Comment (shubha-rajan) [reply dlp/src/main/java/dlp/snippets/RiskAnalysisKMap.java]: replaced with a for loop in 0e72a97

**Comment by kurtisvg** [comment]
forcing merge - (broken tests tracking in #1479)

**Pull request merged**